### ISSUE TITLE
auto-sklearn for clustering

### ISSUE THREAD

**Issue by liliblu** [automl/auto-sklearn#753]
Hello,

I was wondering if anyone is working on an extension for clustering? I was working on this myself, and was thinking of submitting it as a contribution to auto-sklearn. 

I'm brand new to contributing to open source projects so I am not sure where to start, or if you would be interested. 

Thanks,
Lili

**Comment by mfeurer** [comment]
Hi Lili, thanks for you interest in contributing to Auto-sklearn. One of the reason why there is not AutoSklearnClustering so far is the fact that we're not aware of any published work explaining which metric would be useful to tune (and the other reason being maintenance overhead). Practical reasons against using AutoML for "regular" clustering are given in http://www.jmlr.org/proceedings/papers/v27/luxburg12a/luxburg12a.pdf and https://drive.google.com/file/d/0BzRGLkqgrI-qY3FFZjZfMWM2VGc/view Maybe a human-in-the-loop approach such ashttps://arxiv.org/pdf/1602.06886.pdf would be better suited here?

However, I'd be happy to learn about counter examples.

**Comment by liliblu** [comment]
Thanks for your answer, I will look into those sources! 
I also made a python package and snakemake pipeline for this here: https://github.com/liliblu/hypercluster
In case anyone else stumbles upon this and is interested in trying it.
